### PR TITLE
only warn on missing `json` extension

### LINKED ISSUE #2787: [Bug]: [Heroku] [Stack: heroku-24] Previously working deployments suddenly stopped with "ERROR: Required PHP extension 'json' not found."
### Bug report

We've been running the heroku-24 stack with PHP 8.3 and able to deploy no problem for over a month.  Starting around Aug 7th or 8th, deployments stopped working due to dd-trace issue below:

```
-----> Downloading Datadog Agent 7.55.1-1
       Reading package lists...
       Building dependency tree...
       Recommended packages:
         datadog-signing-keys
       The following NEW packages will be installed:
         datadog-heroku-agent
       0 upgraded, 1 newly installed, 0 to remove and 26 not upgraded.
       Need to get 114 MB of archives.
       After this operation, 495 MB of additional disk space will be used.
       Get:1 https://apt.datadoghq.com/ stable/7 amd64 datadog-heroku-agent amd64 1:7.55.1-1 [114 MB]
       Fetched 114 MB in 1s (200 MB/s)
       Download complete and in download only mode
W: --force-yes is deprecated, use one of the options starting with --allow instead.
-----> Installing Datadog Agent
-----> DD_PROCESS_AGENT not set. Removing the process agent.
-----> Installing Datadog runner
-----> Installing Datadog agent wrapper
-----> Datadog PHP extensions app detected
-----> Preparing to install Datadog PHP extensions
       Using 'latest' dd-trace-php release (default).
-----> Downloading installer from https://github.com/DataDog/dd-trace-php/releases/latest/download/datadog-setup.php
-----> Running Datadog installer
       Profiling will be enabled
       Appsec will be enabled
       Searching for available php binaries, this operation might take a while.
       Downloading installable archive from https://github.com/DataDog/dd-trace-php/releases/download/1.2.0/dd-library-php-1.2.0-x86_64-linux-gnu.tar.gz.
       This operation might take a while.
       ...................
       Download completed
       
       Installed required source files to '/tmp/build_05c969f4/opt/dd-library/1.2.0'
       Installing to binary: php (/tmp/build_05c969f4/.heroku/php/bin/php)
       Installing extension to /app/.heroku/php/lib/php/extensions/no-debug-non-zts-20230831
       Copied '/tmp/dd-install/dd-library-php/trace/ext/20230831/ddtrace.so' to '/app/.heroku/php/lib/php/extensions/no-debug-non-zts-20230831/ddtrace.so'
       Copied '/tmp/dd-install/dd-library-php/profiling/ext/20230831/datadog-profiling.so' to '/app/.heroku/php/lib/php/extensions/no-debug-non-zts-20230831/datadog-profiling.so'
       Copied '/tmp/dd-install/dd-library-php/appsec/ext/20230831/ddappsec.so' to '/app/.heroku/php/lib/php/extensions/no-debug-non-zts-20230831/ddappsec.so'
       Created INI file '/app/.heroku/php/etc/php/conf.d/98-ddtrace.ini'
       Installation to 'php (/tmp/build_05c969f4/.heroku/php/bin/php)' was successful
       Installing to binary: php-fpm (/tmp/build_05c969f4/.heroku/php/sbin/php-fpm)
       Killed
       PHP Deprecated:  explode(): Passing null to parameter #2 ($string) of type string is deprecated in /tmp/codon/tmp/buildpacks/766a320daf1a9dc2c1b3c389285af80d17945984/datadog-setup.php on line 1106
       ERROR: Required PHP extension 'json' not found.
       
 !     Push rejected, failed to compile Datadog PHP extensions app.
 !     Push failed
```

PHP version, profiler version and other outputs are from the latest running production dyno. I haven't yet been able to find a way to get that info from the dyno that is being built and is generating the above output including the error. 

From the error message, it would seem that line 1104 below is returning null unexpectedly:

```
$extensions = shell_exec("$binary -m");
```

I'm also reporting this to Heroku as I don't see any indication that there has been a code change related to this in the buildpack, but logging it here in case others are also encountering the same issue.

### PHP version

8.3.10

### Tracer or profiler version

1.2.0

### Installed extensions

php -m
[PHP Modules]
bz2
Core
ctype
curl
datadog-profiling
date
ddappsec
ddtrace
dom
fileinfo
filter
hash
iconv
json
libxml
mbstring
mysqli
mysqlnd
openssl
pcre
PDO
pdo_mysql
pdo_pgsql
pgsql
Phar
posix
random
readline
Reflection
session
SimpleXML
sockets
SPL
standard
tokenizer
xml
xmlreader
xmlwriter
Zend OPcache
zip
zlib

[Zend Modules]
Zend OPcache
datadog-profiling
ddappsec
ddtrace

### Output of `phpinfo()`

_No response_

### Upgrading from

_No response_

### PR DESCRIPTION
**Pull request by realFlowControl**
### Description

Fixes #2787

### Reviewer checklist
- [ ] Test coverage seems ok.
- [ ] Appropriate labels assigned.

### CODE REVIEW

File: datadog-setup.php [hunk 1]
```diff
@@ -1099,12 +1099,16 @@ function check_library_prerequisite_or_exit($requiredLibrary)
...
+function check_php_ext_prerequisite($binary, $extName)
 {
     $extensions = shell_exec(escapeshellarg($binary) . " -m");
 
+    // See: https://github.com/DataDog/dd-trace-php/issues/2787
+    if ($extensions == null || strpos($extensions, '[PHP Modules]') === false) {
+        echo "WARNING: The output of '$binary -m' could not be reliably checked. Please make sure you have the PHP extension '$extName' installed.\n";
+    }
     if (!in_array($extName, array_map("trim", explode("\n", $extensions)))) {
-        print_error_and_exit("Required PHP extension '$extName' not found.\n");
```
Comment (bwoebi) [inline datadog-setup.php:1107]: I think it's still fine to fail _if_ the command actually goes through?

Comment (realFlowControl) [reply datadog-setup.php:1107]: Fair point

Comment (realFlowControl) [reply datadog-setup.php:1107]: done

**Review by iamluc — APPROVED** [review]

**Review by bwoebi — APPROVED** [review]
LGTM :-)

**Pull request merged**